.github/workflows/build_images.yml

```
@@ -0,0 +1,93 @@
+name: build-images
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  actions: read
+  checks: read
+  contents: write
+  deployments: none
+  issues: none
+  packages: none
+  pull-requests: write
+  repository-projects: write
+  security-events: none
+  statuses: none
+
+jobs:
+  build-images:
+    name: Build docker images
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
```

try `contents: write` here job permissions are overwriting top level perms
```suggestion
      contents: write
```